gemma-core/src/main/java/ubic/gemma/core/genome/gene/service/GeneSearchServiceImpl.java

```diff
@@ -161,14 +160,13 @@ public GeneSearchServiceImpl( SearchService searchService, SecurityService secur
          * GET GENES AND GENE SETS
          */
         // SearchSettings settings = SearchSettings.geneSearch( query, taxon );
-        SearchSettings settings = SearchSettings.Factory.newInstance();
-        settings.setQuery( query );
-        settings.noSearches();
-        settings.setSearchGenes( true ); // add searching for genes
-        settings.setSearchGeneSets( true ); // add searching for geneSets
-        settings.setMaxResults( maxGeneralSearchResults );
-        if ( taxon != null )
-            settings.setTaxon( taxon ); // this doesn't work yet
+        SearchSettings settings = SearchSettings.builder()
+                .query( query )
+                .resultType( Gene.class )
+                .resultType( GeneSet.class )
+                .maxResults( maxGeneralSearchResults )
+                .taxon( taxon ) // FIXME: this doesn't work yet
+                .build();
```

Comment:
@ppavlidis Just to give you an idea of how nicer our search settings API is lombok `@Builder`